### DISCUSSION TITLE
Matchit not working

### DISCUSSION THREAD

**Discussion by resting** [JetBrains/ideavim#660] [General]
Trying to enable matchit.

Below is my ~/.ideavimrc

```
let mapleader=','
set surround
set easymotion

nmap <Leader> <Plug>(easymotion-prefix)
nmap <space> <Plug>(easymotion-jumptoanywhere)

packadd matchit
# Plug 'vim-matchit'
# Plug 'chrisbra/matchit'
set matchit
# runtime macros/matchit.vim
```

Have tried various syntaxes but none were working.

That is, nothing happens when pressing % in command mode.

What am I missing?

**Comment by citizenmatt** [comment]
Did you get this working? The config looks correct to me. Where were you testing this?

**Reply by MostHated** [reply]
Did *you* get this working? I am not having luck either.

```vim
" runtime macros/matchit.vim
Plug 'vim-matchit'
" Plug 'chrisbra/matchit' " % to jump between matching tags 
set matchit
let b:match_words = '#region:#endregion'
" let b:match_words = '<:>,<tag>:</tag>,#region:#endregion'
```

**Reply by citizenmatt** [reply]
Have _I_ got it working? Yes! Any of the syntaxes work to enable the extension, and you can always check with `set matchit?` If it shows `nomatchit`, then it's not enabled, `matchit` means it's enabled. Now `%` in supported file types will jump between begin/end pairs. 

Where you might be having problems is with the `b:match_words` variable. Looks like the IdeaVim extension doesn't support this, but has a hardcoded set of begin/end pairs for different file types. I've created an issue to track this: [VIM-3796](https://youtrack.jetbrains.com/issue/VIM-3796).

Alternatively, IdeaVim is open source. If you fancy having a go and submitting a PR, then [the `Matchit.kt` file](https://github.com/JetBrains/ideavim/blob/master/src/main/java/com/maddyhome/idea/vim/extension/matchit/Matchit.kt) is the implementation, and there are a [number of tests in this folder](https://github.com/JetBrains/ideavim/tree/master/src/test/java/org/jetbrains/plugins/ideavim/extension/matchit).

**Reply by MostHated** [reply]
Oh, my apologies. I didn't realize you were with Jetbrains. For some reason, I thought you were another user trying to get it figured out as well. I do see that there are a few default items in the `createCPatterns`, just not the one in particular I was after. 
I may try and have a go at it if I find a bit of spare time.
